test/unit/math/prim/fun/spaced_array_test.cpp

```diff
@@ -0,0 +1,52 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+
+TEST(MathFunctions, spaced_array) {
+  using stan::math::spaced_array;
+
+  std::vector<double> u0 = spaced_array(0, 1, 2);
```

Comment:
[optional]

With this kind of function, it can be helpful to write a general test function, as in:

```
void expect_spaced_array(int K, double low, double high, const std::vector<double>& expected) {
  std::vector<double> found = spaced_array(K, low, high);
  expect_float_eq_array(expected, found);
}
```

Where the final test is a general test to compare arrays.  We should add that ASAP to avoid all this redundancy.  it could go in with the matrix comparisons in

```
test/unit/math/prim/fun/expect_matrix_eq.hpp
```

Reply:
I've added `expect_std_vector_eq` to the header. With that I could simplify most array tests, and implement your suggesion for the `spaced_array` tests.